presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -913,10 +918,11 @@ private Scope legacyVisitQuerySpecification(QuerySpecification node, Optional<Sc
             // to pass down to analyzeFrom
 
             Scope sourceScope = analyzeFrom(node, scope);
+            Scope sourceScopeWithWindow = createWindowScope(node, Optional.of(sourceScope));
```

Comment:
I checked what is supported in Postgres. If we have a table created as ``` CREATE TABLE sample(a bigint, b bigint);```, then the following statement works:
```SELECT sum(s) OVER w sum_s FROM (SELECT b, sum(a) over (partition by b) s FROM sample) AS sumofa WINDOW w AS (PARTITION BY b);``` But this does not work:
```
SELECT sum(s) OVER w sum_s FROM (SELECT b, sum(a) over w s FROM sample) AS sumofa WINDOW w AS (PARTITION BY b);
ERROR:  window "w" does not exist
LINE 1: ...T sum(s) OVER w sum_s FROM (SELECT b, sum(a) over w s FROM s...
``` 
This indicates that the window definitions present in the outer query are not passed down to the inner subquery. This is also how I interpreted the SQL standard. So I'm going to remove the code that retrieves definition from the parent scope. As the code stands today, the parent scope won't have the window definition in the first place, since I'm not passing down the window definition to the scope in `analyzeFrom`.